### PR TITLE
Proofread and improve the optimization guides

### PR DESCRIPTION
**Pull request by Calinou**
This improves writing style and simplifies sentences and vocabulary a bit. Important parts have also been boldened to make the text easier to go through.

### CODE REVIEW

File: tutorials/optimization/batching.rst [hunk 1]
```diff
@@ -141,221 +140,226 @@ balance the costs and benefits in your project.
...
 
 Light scissoring is controlled with the :ref:`scissor_area_threshold
 <class_ProjectSettings_property_rendering/batching/lights/scissor_area_threshold>`
 project setting. This value is between 1.0 and 0.0, with 1.0 being off (no
 scissoring), and 0.0 being scissoring in every circumstance. The reason for the
 setting is that there may be some small cost to scissoring on some hardware.
-Generally though, when you are using lighting, it should result in some
-performance gains.
+That said, scissoring should usually result in performance gains when you're
+using 2D lighting.
```
Comment (clayjohn) [inline tutorials/optimization/batching.rst:240]: "For instance"

File: tutorials/optimization/cpu_optimization.rst [hunk 2]
```diff
@@ -21,98 +21,102 @@ CPU profilers
...
 
 When manually timing functions, it is usually a good idea to run the function
-many times (say ``1000`` or more times), instead of just once (unless it is a
-very slow function). A large part of the reason for this is that timers often
-have limited accuracy, and CPUs will schedule processes in a haphazard manner,
-so an average over a series of runs is more accurate than a single measurement.
+many times (1,000 or more times), instead of just once (unless it is a very slow
+function). The reason for doing this is that timers often have limited accuracy.
+Moreover, CPUs will schedule processes in a haphazard manner. Therefore, an
+average over a series of runs is more accurate than a single measurement.
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:119]: CPU caches are something ...

File: tutorials/optimization/cpu_optimization.rst [hunk 3]
```diff
@@ -186,73 +193,82 @@ Third parties provide support for several other languages, including `Rust
 C++
 ~~~
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:195]: "Threads" typo

File: tutorials/optimization/cpu_optimization.rst [hunk 4]
```diff
@@ -186,73 +193,82 @@ Third parties provide support for several other languages, including `Rust
...
 Physics
 =======
 
-In some situations physics can end up becoming a bottleneck, particularly with
-complex worlds, and large numbers of physics objects.
+In some situations, physics can end up becoming a bottleneck. This is
+particularly the case with complex worlds and large numbers of physics objects.
 
-Some techniques to speed up physics:
+Here are some techniques to speed up physics:
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:250]: Shouldnt it just be a single "_" at the end of the link? I may be wrong, feel free to ignore.

File: tutorials/optimization/optimizing_3d_performance.rst [hunk 5]
```diff
@@ -22,57 +22,64 @@ in the street you are in, as well as the sky and a few birds flying overhead. As
 far as a naive renderer is concerned however, you can still see the entire town.
 It won't just render the buildings in front of you, it will render the street
 behind that, with the people on that street, the buildings behind that. You
-quickly end up in situations where you are attempting to render 10x, or 100x
-more than what is visible.
+quickly end up in situations where you are attempting to render 10× or 100× more
+than what is visible.
```
Comment (clayjohn) [inline tutorials/optimization/optimizing_3d_performance.rst:27]: This paragraph refers to the paragraph that used to be directly above and should be moved back up. The added paragraph above can go under this one just fine as it is application to more than just PVS

File: tutorials/optimization/index.rst [hunk 6]
```diff
@@ -4,33 +4,33 @@ Optimization
...
-   material to reduce the costs, but at the same time sorting has a cost.
+   material to reduce the costs. At the same time, sorting has a cost.
 
--  In 3D physics a similar situation happens. The best algorithms to
+-  In 3D physics, a similar situation happens. The best algorithms to
    handle large amounts of physics objects (such as SAP) are slow at
-   insertion/removal of objects and ray-casting. Algorithms that allow faster
-   insertion and removal, as well as ray-casting, will not be able to handle as
+   insertion/removal of objects and raycasting. Algorithms that allow faster
+   insertion and removal, as well as raycasting, will not be able to handle as
```
Comment (clayjohn) [inline tutorials/optimization/index.rst:24]: This paragraph and the next go over 80 columns if you care

File: tutorials/optimization/gpu_optimization.rst [hunk 7]
```diff
@@ -1,75 +1,76 @@
...
-examining changes in frame rate.
+situation on the CPU. This is because, often, you can only change performance
+indirectly by changing the instructions you give to the GPU. Also, it may be
+more difficult to take measurements. In many cases, the only way of measuring
+performance is by examining changes in the time spent rendering each frame.
 
-Drawcalls, state changes, and APIs
-==================================
+Draw calls, state changes, and APIs
+==================================~
```
Comment (clayjohn) [inline tutorials/optimization/gpu_optimization.rst:26]: I think this should be another "="

File: tutorials/optimization/general_optimization.rst [hunk 8]
```diff
@@ -165,67 +166,71 @@ Performant design
...
+As with design, you should concentrate your efforts first on making sure the
 algorithms and data structures are the best they can be. Data access should be
 local (to make best use of CPU cache), and it can often be better to use compact
-storage of data (again, always profile to test results). Often you precalculate
-heavy computations ahead of time (e.g. at level load, or loading precalculated
-data files).
+storage of data (again, always profile to test results). Often, you precalculate
+heavy computations ahead of time. This can be done by performing the computation
+when loading a level, by loading a file containing precalculated data or simply
+by storing the results of complex calculations into a script constant and
```
Comment (clayjohn) [inline tutorials/optimization/general_optimization.rst:224]: add backticks around "A"

File: tutorials/optimization/cpu_optimization.rst [hunk 9]
```diff
@@ -121,21 +125,21 @@ optimization is working (or not).
 Caches
 ======
 
-Something else to be particularly aware of, especially when comparing timing
-results of two different versions of a function, is that the results can be
+Caches are something else to be particularly aware of, especially when comparing
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:128]:
```suggestion
CPU caches are something else to be particularly aware of, especially when comparing
```

File: tutorials/optimization/cpu_optimization.rst [hunk 10]
```diff
@@ -186,73 +193,82 @@ Third parties provide support for several other languages, including `Rust
...
 
 The disadvantage of threads is that you have to be incredibly careful. As each
 CPU core operates independently, they can end up trying to access the same
 memory at the same time. One thread can be reading to a variable while another
-is writing. Before you use threads make sure you understand the dangers and how
-to try and prevent these race conditions.
+is writing: this is called a *race condition*. Before you use threads, make sure
+you understand the dangers and how to try and prevent these race conditions.
+
+Threasd can also make debugging considerably more difficult. The GDScript
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:216]:
```suggestion
Threads can also make debugging considerably more difficult. The GDScript
```

File: tutorials/optimization/cpu_optimization.rst [hunk 11]
```diff
@@ -186,73 +193,82 @@ Third parties provide support for several other languages, including `Rust
...
 
-The solution to this problem is 'fixed timestep interpolation', which involves
+The solution to jitter is to use *fixed timestep interpolation*, which involves
 smoothing the rendered positions and rotations over multiple frames to match the
-physics. You can either implement this yourself or use a third-party addon.
-Interpolation is a very cheap operation, performance wise, compared to running a
-physics tick, orders of magnitude faster, so this can be a significant win, as
-well as reducing jitter.
+physics. You can either implement this yourself or use a
+`third-party addon <https://github.com/lawnjelly/smoothing-addon>`__.
```
Comment (clayjohn) [inline tutorials/optimization/cpu_optimization.rst:271]: I thought hyperlinks just had a single "_" at the end. I could be wrong though. If i am, please ignore.

File: tutorials/optimization/general_optimization.rst [hunk 12]
```diff
@@ -247,14 +252,14 @@ function 'A', then optimizing A can have a massive effect on performance.
...
 
-    A: 1 ms 
-    Everything else: 1ms 
+    A: 1 ms
+    Everything else: 1ms
     Total frame time: 2 ms
 
-So in this example improving this bottleneck A by a factor of 9x, decreases
-overall frame time by 5x, and increases frames per second by 5x.
+In this example, improving this bottleneck A by a factor of 9× decreases
```
Comment (clayjohn) [inline tutorials/optimization/general_optimization.rst:259]:
```suggestion
In this example, improving this bottleneck ``A`` by a factor of 9× decreases
```

File: tutorials/optimization/batching.rst [hunk 13]
```diff
@@ -141,221 +140,226 @@ balance the costs and benefits in your project.
...
 
-However, within a single draw call (batch) we cannot change uniforms. This means
-that naively, we would not be able to batch together items or commands that
-change final_modulate, or item transform. Unfortunately that is an awful lot of
-cases. Sprites for instance typically are individual nodes with their own item
-transform, and they may have their own color modulate.
+However, within a single draw call (batch), we cannot change uniforms. This
+means that naively, we would not be able to batch together items or commands
+that change ``final_modulate`` or an item's transform. Unfortunately, that
+happens in an awful lot of cases. For instances, sprites are typically
```
Comment (clayjohn) [inline tutorials/optimization/batching.rst:270]:
```suggestion
happens in an awful lot of cases. For instance, sprites are typically
```

File: tutorials/optimization/gpu_optimization.rst (hunk 7 above)
Comment (clayjohn) [inline tutorials/optimization/gpu_optimization.rst:26]:
```suggestion
===================================
```

File: tutorials/optimization/optimizing_3d_performance.rst [hunk 14]
```diff
@@ -22,57 +22,64 @@ in the street you are in, as well as the sky and a few birds flying overhead. As
...
 you have to do is work out when your viewer is in street ``A`` (perhaps using
 Godot Areas), then you can hide the other streets.
 
-This is a manual version of what is known as a 'potentially visible set'. It is
+In some cases, you may have to adapt your level design to add more occlusion
+oppotunities. For example, you may have to add more walls to prevent the player
+from seeing too far away, which would decrease performance due to the lost
+opportunies for occlusion culling.
+
+This is a manual version of what is known as a "potentially visible set". It is
```
Comment (clayjohn) [inline tutorials/optimization/optimizing_3d_performance.rst:48]: This paragraph refers to the paragraph that used to be directly above and should be moved back up. The added paragraph above can go under this one just fine as it is application to more than just PVS

File: tutorials/optimization/cpu_optimization.rst (hunk 11 above)
Comment (Calinou) [reply tutorials/optimization/cpu_optimization.rst:271]: Both can be used, depending on your goals. A single underscore creates a named reference (you can't duplicate them without causing a Sphinx build warning). A double underscore creates an anonymous reference (you can duplicate them without causing a Sphinx build warning).

Due to this, I prefer using anonymous references unless they can't be used for some reason.

**Comment by Calinou** [comment]
I fixed some of the cases where the line length got past 80 characters. That said, it can't always be done (specifically for link texts or inline code).

**Comment by akien-mga** [comment]
Needs rebase and a final amend based on @clayjohn's review.

File: tutorials/optimization/cpu_optimization.rst [hunk 15]
```diff
@@ -186,73 +194,83 @@ Third parties provide support for several other languages, including `Rust
...
 ~~~
 
-Godot is written in C++. Using C++ will usually result in the fastest code,
-however, on a practical level, it is the most difficult to deploy to end users'
-machines on different platforms. Options for using C++ include GDNative, and
-custom modules.
+Godot is written in C++. Using C++ will usually result in the fastest code.
+However, on a practical level, it is the most difficult to deploy to end users'
+machines on different platforms. Options for using C++ include
+:ref:`toc-tutorials-gdnative` and
```
[the remark this answers is not in the document]
Comment (Calinou) [reply tutorials/optimization/cpu_optimization.rst]: I removed the link for now. I'll add a link to the "What is GDNative?" page once it's merged.

File: tutorials/optimization/cpu_optimization.rst [hunk 16]
```diff
@@ -165,16 +169,20 @@ code, consider moving those calculations to a faster language.
 GDScript
 ~~~~~~~~
 
-GDScript is designed to be easy to use and iterate, and is ideal for making many
-types of games. However, ease of use is considered more important than
-performance, so if you need to make heavy calculations, consider moving some of
-your project to one of the other languages.
+:ref:`toc-learn-scripting-gdscript` is designed to be easy to use and iterate,
```
Comment (akien-mga) [inline tutorials/optimization/cpu_optimization.rst:172]: There's an error on this one but I don't get why:
```
/home/runner/work/godot-docs/godot-docs/tutorials/optimization/cpu_optimization.rst:172:undefined label: toc-learn-scripting-gdscript (if the link has no caption the label must precede a section header)
```

Comment (Calinou) [reply tutorials/optimization/cpu_optimization.rst]: I tried adding explicit names for TOC references, does it work now?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**